source/slang/slang-ir-dce.cpp

```diff
@@ -256,6 +256,11 @@ struct DeadCodeEliminationContext
             }
         }
 
+        if (options.keepLayoutsAlive && inst->findDecoration<IRLayoutDecoration>())
+        {
+            return true;
+        }
+
```

Comment:
As mentioned in the discussion thread for the PR, I'm not 100% happy with this, but I'll concede that it is not a large amount of code, so it is fine as the solution for now.

The "original sin" here is really that we don't have a singular instruction that represents the `ProgramLayout` at the IR level, because *that* is what we'd want to keep alive.

In the future it might be better if we had an `IRProgramLayout` instruction, which in turn referenced the layouts for global parameters, entry points, etc. In that model, the linking process would start with the `IRProgramLayout` for what we want to codegen (or maybe one entry point within it...), which would ensure that all the things being laid out get linked, and then *after* linking we would set about transferring layout information from the centralized `IRProgramLayout` down to the entry points, global parameters, etc. (similar to how we wait until after linking to copy information from entry points to their parameters today).